### ISSUE TITLE
2.3b8: 4.4-subscription-ops.test fails with server-error-internal-error (Invalid group tag.)

### ISSUE THREAD

**Issue by OdyX** [apple/cups#5576]
Thanks for the 2.3b8 version.

Unfortunately, in a Debian `sid` chroot, the `4.4-subscription-ops.test` fails with the following log:

```
[07/May/2019:08:05:00 +0000] "4.4-subscription-ops.test":
    Add Printer Subscription w/Lease                                     [PASS]
        notify-subscription-id (integer) = 3
    Verify Subscription Expiration                                       [PASS]
    Add 2 Printer Subscriptions w/Lease                                  [FAIL]
        RECEIVED: 0 bytes in response
        status-code = server-error-internal-error (Invalid group tag.)
        IPP request failed with status server-error-internal-error (Invalid group tag.)
    List Printer Subscriptions                                           [PASS]
        notify-events (keyword) = printer-state-changed
        notify-printer-uri (uri) = ipp://localhost:8631/printers/Test1
        notify-subscription-id (integer) = 4
    Check MaxSubscriptions limits                                        [PASS]
```

I can reliably reproduce it with the following minimal build-script:
```bash
#!/bin/sh
set -e
export LANG=C
aclocal
autoconf
./configure --build=x86_64-linux-gnu \
        --prefix=/usr \
        --includedir=\${prefix}/include \
        --mandir=\${prefix}/share/man \
        --infodir=\${prefix}/share/info \
        --sysconfdir=/etc \
        --localstatedir=/var \
        --disable-silent-rules \
        --libdir=\${prefix}/lib/x86_64-linux-gnu \
        --libexecdir=\${prefix}/lib/x86_64-linux-gnu \
        --disable-maintainer-mode \
        --disable-dependency-tracking \
        --with-docdir=/usr/share/cups/doc-root \
        --localedir=/usr/share/cups/locale \
        --enable-libpaper \
        --enable-ssl \
        --enable-gnutls \
        --enable-threads \
        --enable-static \
        --enable-debug \
        --enable-dbus \
        --with-dbusdir=/etc/dbus-1 \
        --enable-gssapi \
        --enable-avahi \
        --disable-launchd \
        --with-cups-group=lp \
        --with-system-groups=lpadmin \
        --with-rundir=/run/cups \
        --with-printcap=/run/cups/printcap \
        --with-log-file-perm=0640 \
        --with-local_protocols='dnssd' \
        --with-systemd=/lib/systemd/system \
        --enable-libusb
make -j16
make check
```

I tried to bisect from v2.3b7 to v2.3b8 and it points to feb13eeaf60f7f3823d648c1f97a263b17462552, which seems weird, but I can confirm with the above script that building on its immediate parent (f1ac9f5889e6a9a8996041ab46e6ca1e0cb8a238) has that test working, while feb13eeaf60f7f3823d648c1f97a263b17462552 hasn't. 

To be fair, on f1ac9f5889e6a9a8996041ab46e6ca1e0cb8a238, the log is noisy, but the test passes:
```
Running IPP compliance tests...
Performing 4.1-requests.test: PASS
Performing 4.2-cups-printer-ops.test: PASS
Performing 4.3-job-ops.test: ipptool: Filename "testfile.ps" (mapped to "testfile.ps") on line 27 of "4.3-job-ops.test" cannot be read.
PASS
Performing 4.4-subscription-ops.test: PASS
Performing ipp-2.1.test: PASS
```

**Comment by michaelrsweet** [comment]
Not sure what is going wrong here since I did a "make check" test on macOS as well as Fedora and Ubuntu current before I did the beta tarball...

Investigating...

**Comment by michaelrsweet** [comment]
Cannot reproduce with CUPS master on Ubuntu 19.04.

@OdyX Can you confirm you are using stock CUPS? The configure options you list there aren't all part of the stock configure script. Also, running aclocal MAY break things in subtle ways (there is a reason we don't use it!), and unless you are changing configure.ac or the m4 files in config-scripts there is no reason to run autoconf (and if you are make sure to use the "-f" option to force an update since by default autoconf only checks that configure.ac is newer than configure and not the included files...)

**Comment by OdyX** [comment]
I have now worked on automating Debian `sid` builds on Travis-CI, in docker containers. See https://github.com/OdyX/cups/blob/master-travis/.travis.yml and the resulting builds on the `master-travis` branch: https://travis-ci.org/OdyX/cups/branches.

I have prepared a build matrix to test various configure flags combination, but kept these "standard" ones:
```
--build=x86_64-linux-gnu --prefix=/usr --includedir=\${prefix}/include --mandir=\${prefix}/share/man --infodir=\${prefix}/share/info --sysconfdir=/etc --localstatedir=/var --libdir=\${prefix}/lib/x86_64-linux-gnu --libexecdir=\${prefix}/lib/x86_64-linux-gnu --with-docdir=/usr/share/cups/doc-root --localedir=/usr/share/cups/locale 
```

Specifically, this build fails on `master` for the two combinations of configure flags I gave: https://travis-ci.org/OdyX/cups/builds/529321998

So yes, I can confirm that stock CUPS `master`, directly from the repository, fails in minimal Debian `sid` chroots/containers.

**Comment by michaelrsweet** [comment]
OK, looking at the docker container logs I'm seeing:

    Unable to open listen socket for address [v1.::1]:8631 - Cannot assign requested address.
    [Job N] Unable to open print file: No such file or directory
    (lots of these)

This tells me there is a permissions problem with the docker environment, plus IPv6 isn't being supported.

**Comment by OdyX** [comment]
It works on 2.3b7 without issues, after ignoring the `Unable to open listen socket for address` error log entry: https://travis-ci.org/OdyX/cups/jobs/529648064

**Comment by michaelrsweet** [comment]
Merged fix from PR #5577...
